docs/docs/consumer-development/changeover-procedure.md

```diff
@@ -0,0 +1,231 @@
+        // e.g. "upgrade_height": 100 => "revision_height": 101
+        "revision_height": 1,
+    },
+...
+}
+```
+RevisionNumber: 0, RevisionHeight: 111
+:::
+
+* `genesis_hash` can be safely ignored because the chain is already running. A hash of the sovereign chain's initial genesis may be used
+
+* `binary_hash` may not be available ahead of time. All chains performing the changeover go through rigorous testing - if bugs are caught and fixed the hash listed in the proposal may not be the most recent one.
+
+* `spawn_time` listed in the proposal MUST be before the `upgrade_height` listed in the the upgrade proposal on the sovereign chain.
+:::caution
+`spawn_time` must occur before the `upgrade_height` on the sovereign chain is reached becasue the `provider` chain must generate the `ConsumerGenesis` that contains the **validator set** that will be used after the changeover.
+:::
+
+* `unbonding_period` must correspond to the value used on the sovereign chain. Otherwise, the clients used for the ccv protocol may be incorrectly initialized.
+
+* `distribution_transmission_channel` **should be set**.
+
+:::note
+Populating `distribution_transmission_channel` will enable the sovereign chain to re-use one of the existing channels to the provider for consumer chain rewards distribution. This will preserve the `ibc denom` that may already be in use.
+
+If the parameter is not set, a new channel will be created.
+:::
+
+* `ccv_timeout_period` has no important notes
+
+* `transfer_timeout_period` has no important notes
+
+* `consumer_redistribution_fraction` has no important notes
+
+* `blocks_per_distribution_transmission` has no important notes
+
+* `historical_entries` has no important notes
+
+
+### 2. upgrade proposal on sovereign chain
+
+The sovereign chain creates an upgrade proposal to include the `interchain-security/x/ccv/consumer` module.
+
+:::caution
+The upgrade height in the proposal should correspond to a height that is after the `spawn_time` in the consumer addition proposal submitted to the `provider` chain.
+:::
+
+Otherwise, the upgrade is indistinguishable from a regular on-chain upgrade proposal.
+
+### 3. spawn time is reached
+
+When the `spawn_time` is reached on the `provider` it will generate a `ConsumerGenesis` that contains the validator set that will supercede the `sovereign` validator set.
+
+This `ConsumerGenesis` must be available on the sovereign chain during the on-chain upgrade.
+
+### 4. sovereign chain upgrade
+
+Performing the on-chain upgrade on the sovereign chain will add the `ccv/consumer` module and allow the chain to become a `consumer` of replicated security.
+
+:::caution
+The `ConsumerGenesis` must be exported to a file called `genesis.json` and placed in the following folder on the sovereign chain before the upgade: `<CURRENT_USER_DIR>/.sovereign/config/genesis.json` (`<CURRENT_USER>` is the home directory of the user executing the chain process)
```

Comment:
I think this is only true of Stride and the dummy app, but overall the JSON doesn't have to be configured this way. Maybe I'm wrong tho

Reply:
Oh yeah. We'd need to actually write instructions on how to perform the upgrade.